src/classes/ADDR_Account_TDTM.cls

```diff
@@ -108,12 +108,19 @@ public with sharing class ADDR_Account_TDTM extends TDTM_Runnable {
         if (acc1 == null) 
            return false;
         
-        return (
+        boolean isSpecified = 
            acc1.BillingStreet != null ||
            acc1.BillingCity != null ||
            acc1.BillingState != null ||
            acc1.BillingPostalCode != null ||
-           acc1.BillingCountry != null);
+            // note that we decided to ignore country so that a default value won't create unnecessary address objects
+           (acc1.BillingCountry != null && !ADDR_Addresses_TDTM.isStateCountryPicklistsEnabled);
+           
```

Comment:
the first case is looking at Billing Country.  the second set is looking at the state code which must be done dynamically.  yes, the comment is twice on purpose to explain these two special cases.